src/main/java/org/jabref/model/groups/WordKeywordGroup.java

```diff
@@ -113,6 +118,12 @@ public boolean contains(BibEntry entry) {
 
     private Set<String> getFieldContentAsWords(BibEntry entry) {
         if (onlySplitWordsAtSeparator) {
+            if (BibEntry.TYPE_HEADER.equals(searchField)) {
+                Optional<EntryType> entryType = EntryTypes.getType(entry.getType(), BibDatabaseMode.BIBLATEX);
+                if (entryType.isPresent()) {
```

Comment:
As above, `entryType = EntryTypes.getType(entry.getType(), BibDatabaseMode.BIBLATEX).orElse(entry.getType())` is slightly better.